### PR TITLE
Packet queue optimization

### PR DESCRIPTION
**Pull request by kirill-scherba**
- the index map added to the send and receive queue to optimize "Find By Id" function
- old "write queue" code removed
- some minor changes in code and comments

### CODE REVIEW

File: src/packet_queue.c [hunk 1]
```diff
@@ -34,13 +34,15 @@
...
 // Local functions
-
 #ifdef RESERVED
 static trudpPacketQueueData *_trudpPacketQueueAddTime(trudpPacketQueue *tq,
         void *packet, size_t packet_length, uint64_t expected_time);
 trudpPacketQueueData *trudpPacketQueueFindByTime(trudpPacketQueue *tq, uint64_t t);
 #endif
 
+// Local macros
+#define PACKET_ID trudpPacketGetId(trudpPacketQueueDataGetPacket(tqd))
```
Comment (Si1ver) [inline src/packet_queue.c]: Let's inline it, it's so short!

**Review by Si1ver — APPROVED** [review]

File: src/packet_queue.c (hunk 1 above)
Comment (chernikovdmitry) [inline src/packet_queue.c]: Баба Яга против!

File: src/packet_queue.c [hunk 2]
```diff
@@ -69,6 +73,7 @@ void trudpPacketQueueDestroy(trudpPacketQueue *tq) {
  * @return Zero at success
  */
 int trudpPacketQueueFree(trudpPacketQueue *tq) {
+    teoMapClear(tq->idx);
```
Comment (chernikovdmitry) [inline src/packet_queue.c:74]: Crash? tq checked for NULL on next line

**Review by chernikovdmitry — CHANGES_REQUESTED** [review]

File: src/packet_queue.c (hunk 2 above)
Comment (Si1ver) [reply src/packet_queue.c:74]: nice catch

**Review by chernikovdmitry — APPROVED** [review]
Much better! Shiny new trudp for the glory of Satan!

**Pull request merged**